.github/workflows/pypi-build.yml

```diff
@@ -0,0 +1,44 @@
+on:
+  pull_request:
+    branches: [ "main" ]
+jobs:
+  pypi-publish:
+    name: Upload release to PyPI
+    runs-on: ubuntu-latest
+    environment:
+      name: pypi
+      url: https://pypi.org/project/causica
+    permissions:
+      id-token: write  # IMPORTANT: this permission is mandatory for trusted publishing
+
+    steps:
+    - uses: actions/checkout@v3
+
+    - uses: actions/setup-python@v2
+      with:
+        python-version: "3.10"
+
+    - uses: actions/cache@v3
```

Comment:
Shall we drop the cache for the publish workflow?

Reply:
sure